### DISCUSSION TITLE
how to split text to different row with different fonts?

### DISCUSSION THREAD

**Discussion by mat-sche** [olikraus/u8g2#1982] [Q&A] [answered]
Hi, I'm newbie in programming and english langwitch using .... my german is bedder :)

I am using this great graphics library but I have a problem or not have a great idea how to print text  in one step with more text characters than possible aspect ratio. Furthermore have i this also with different fonts
My Display have 64x128 digits.
Example:

      u8g2.setFont(u8g2_font_8x13B_mf);
      u8g2.setCursor(2, 10);
      u8g2.print("Found fingerprint sensorxxxxxxxxxxxxxxxxx!");
      u8g2.sendBuffer();

In this case the sign xxxxx is not printing on display.....
What can I do?

MAT

**Comment by olikraus** [comment, marked as answer]
Maybe the string is too long for your display. Remember: there is no automatic line break

**Reply by mat-sche** [reply]
> Maybe the string is too long for your display

Thats right and I know this. 
> Remember: there is no automatic line break

That be would be great.....
How do I calculate the max. characters per line in relation to the font used?

**Reply by olikraus** [reply]
You can pass a (sub) string and calculate the expected size: https://github.com/olikraus/u8g2/wiki/u8g2reference#getutf8width
Another option would be to draw one char after the other and check whether the max x position is reached.

In the following example we will use the return value of https://github.com/olikraus/u8g2/wiki/u8g2reference#drawglyph to advance the x position:

``` C++
  u8g2_uint_t x = 0; // start position
  u8g2_uint_t y = 20;
  ...
  char *s = "Hello Very Long String";
  while( *s != '\0' ) {
    while( *s != '\0' && x < u8g2.getDisplayWidth() )
       x += u8g2.drawGylph(x, y, *s++ )
    x = 0; y += 10;  // next line
  }
```

**Reply by mat-sche** [reply]
Many thanks for your replay, all information I can find in the reference! Perfect but a lot of information for me, for me as newbee :)! I try to understand that all ;)
